great_expectations/types/__init__.py

```diff
@@ -135,7 +135,7 @@ def to_raw_dict(self):
 
     def to_dict(self) -> dict:
         new_dict = {
-            key: self[key]
+            key: self.get(key)
```

Comment:
Without this:
```python
AttributeError: ‘CategoricalColumnDomainBuilder’ object has no attribute ‘cardinality_checker’
```

We need to ensure we actually have the attr before trying to convert to dict - Alice breaks without this change.

Reply:
@cdkini This could be due to a bad merge -- there is the `cardinality_checker` property now.  Happy to discuss.  Thanks!